api/base/serializers.py

```diff
@@ -448,7 +448,11 @@ def get_url(self, obj, view_name, request, format):
                 else:
                     url = self.reverse(view, kwargs=kwargs, request=request, format=format)
                     if self.filter:
-                        url = '{}?filter{}'.format(url, self.format_filter(obj))
+                        formatted_filter = self.format_filter(obj)
+                        if formatted_filter:
```

Comment:
Can you describe to me the circumstance in which `self.filter` is `True` but `formatted_filter` is `False` and why, when both of these are the case, the url is `None` rather than an unfiltered URL or similar? This may be obvious when I get later in the code, but just in case it's not.